rust/Cargo.toml

```diff
@@ -37,7 +37,7 @@ async-trait = "0.1.60"
 byteorder = "1.4.3"
 chrono = "0.4.23"
 clap = { version = "4.1.1", features = ["derive"], optional = true }
-object_store = { version = "0.5.6", features = ["aws_profile", "gcp"] }
+object_store = { version = "0.6.0", features = ["aws", "gcp"] }
```

Comment:
do we need to make this sync with datafusion / arrow-rs?

Reply:
Yeah you are right, bumping.

Reply:
ok, we need object store `0.6.0` which is used by arrow `40.0`. BUT, datafusion `25.0` is still on `39.0`. going to put this PR on hold till the next datafusion release.

Note: there are some breaking changes. I'll make a separate PR when the new datafusion is released